### PR TITLE
[SYCL] Remove host run and dependencies from SYCL/DeprecatedFeatures tests

### PR DESCRIPTION
**Pull request by steffenlarsen**
This commit removes the host run and any assumptions and operations related to the host device from the tests in SYCL/DeprecatedFeatures.

### CODE REVIEW

File: SYCL/DeprecatedFeatures/FunctionPointers/fp-as-kernel-arg.cpp [hunk 1]
```diff
@@ -34,17 +33,15 @@ int main() {
...
-    // For now, we don't have runtimes that report required OpenCL extension and
-    // it is hard to understand should this functionality be supported or not.
-    // So, let's skip this test if FptrStorage is 0, which means that by some
-    // reason we failed to obtain device function pointer. Just to avoid false
-    // alarms
-    if (0 == FptrStorage) {
-      std::cout << "Test PASSED. (it was actually skipped)" << std::endl;
-      return 0;
-    }
+  // FIXME: update this check with query to supported extension
```
Comment (AlexeySachkov) [inline SYCL/DeprecatedFeatures/FunctionPointers/fp-as-kernel-arg.cpp]: Function pointer tests were removed in #1187

File: SYCL/DeprecatedFeatures/SpecConsts1.2.1/composite-type.cpp [hunk 2]
```diff
@@ -1,7 +1,6 @@
 // UNSUPPORTED: cuda || hip
 //
 // RUN: %clangxx -fsycl -D__SYCL_INTERNAL_API %s -o %t.out
```
Comment (AlexeySachkov) [inline SYCL/DeprecatedFeatures/SpecConsts1.2.1/composite-type.cpp]: Spec constant tests were removed in #1187

**Review by AlexeySachkov — CHANGES_REQUESTED** [review]

File: SYCL/DeprecatedFeatures/FunctionPointers/fp-as-kernel-arg.cpp (hunk 1 above)
Comment (steffenlarsen) [reply SYCL/DeprecatedFeatures/FunctionPointers/fp-as-kernel-arg.cpp]: Removed after merge commit.

File: SYCL/DeprecatedFeatures/SpecConsts1.2.1/composite-type.cpp (hunk 2 above)
Comment (steffenlarsen) [reply SYCL/DeprecatedFeatures/SpecConsts1.2.1/composite-type.cpp]: Removed after merge commit.

**Review by AlexeySachkov — APPROVED** [review]

**Review by dm-vodopyanov — APPROVED** [review]

**Pull request merged**